helpers.go

```diff
@@ -18,25 +18,49 @@ var neverGonnaHappen = time.Hour * 24 * 365 * 100
 var lineSepPosix = "\n"
 var lineSepWindows = "\r\n"
 
+var processExitPollInterval = 10 * time.Millisecond
+var processExitExtraWait = 500 * time.Millisecond
+
 type cmdExit struct {
 	ProcessState *os.ProcessState
 	Err          error
 }
 
 // waitForCmdExit turns process.wait() into a channel so that it can be used within a select{} statement
-func waitForCmdExit(cmd *exec.Cmd) chan cmdExit {
-	exit := make(chan cmdExit, 1)
+func waitForCmdExit(cmd *exec.Cmd) chan *cmdExit {
+	exit := make(chan *cmdExit, 1)
 	go func() {
 		err := cmd.Wait()
-		exit <- cmdExit{ProcessState: cmd.ProcessState, Err: err}
+		exit <- &cmdExit{ProcessState: cmd.ProcessState, Err: err}
 	}()
 	return exit
 }
 
+// ttExited returns a channel that sends the given termtest's command cmdExit info when available.
+// This can be used within a select{} statement.
+// If waitExtra is given, waits a little bit before sending cmdExit info. This allows any fellow
+// switch cases to handle unprocessed stdout.
+func ttExited(tt *TermTest, waitExtra bool) chan *cmdExit {
+	return waitChan(func() *cmdExit {
+		ticker := time.NewTicker(processExitPollInterval)
+		for {
+			select {
+			case <-ticker.C:
+				if tt.exited != nil {
+					if waitExtra {
```

Comment:
Could you explain the rationale for this in a comment?

Reply:
It is explained in the function description. Should I move/copy that comment here?

Reply:
No I understand what the param does. What I'm asking is why is this a thing at all? I see this function is called from two different places, once with this param set to true and once set to false. How did you decide one of them needed this and the other didn't?

Reply:
I added some more comments around this. Basically, you want to wait if you have sibling output consumers that are expecting output so that you don't immediately fail the expect if the process exits. You don't want to wait if you are expecting the process to exit or have exited (i.e. `cp.ExpectExit*()`)